atum/src/test/scala/za/co/absa/atum/ControlInfoToJsonSerializationSpec.scala

```diff
@@ -138,6 +148,61 @@ class ControlInfoToJsonSerializationSpec extends AnyFlatSpec with Matchers {
      |}]
      |}""".stripMargin.filter(_ >= ' ')
 
+  private val exampleInputJsonCustomSoftware: String = s"""{
+    |"metadata":{
+    |"sourceApplication":"FrontArena",
+    |"country":"ZA",
+    |"historyType":"Snapshot",
+    |"dataFilename":"example.dat",
+    |"sourceType":"",
+    |"version":1,
+    |"informationDate":"01-01-2017",
+    |"additionalInfo":{
+    |"key1":"value1",
+    |"key2":"value2"
+    |}
+    |},
+    |"checkpoints":[{
+    |"name":"Source",
+    |"software":"SomeOtherSoftware",
+    |"version":"v10",
+    |"processStartTime":"01-01-2017 08:00:00",
+    |"processEndTime":"01-01-2017 08:00:00",
+    |"workflowName":"Source",
+    |"order":1,
+    |"controls":[{
+    |"controlName":"pvControlTotal",
+    |"controlType":"type.aggregatedTotal",
+    |"controlCol":"pv",
+    |"controlValue":"32847283324.324324"
+    |},{
+    |"controlName":"recordCount",
+    |"controlType":"type.Count",
+    |"controlCol":"id",
+    |"controlValue":243
+    |}]
+    |},{
+    |"name":"Raw",
+    |"software":"$software",
+    |"version":"$version",
+    |"processStartTime":"01-01-2017 08:00:00",
+    |"processEndTime":"01-01-2017 08:00:00",
+    |"workflowName":"Raw",
+    |"order":2,
+    |"controls":[{
+    |"controlName":"pvControlTotal",
+    |"controlType":"type.aggregatedTotal",
+    |"controlCol":"pv",
+    |"controlValue":"32847283324.324324"
+    |},{
+    |"controlName":"recordCount",
+    |"controlType":"type.Count",
+    |"controlCol":"id",
+    |"controlValue":243
+    |}]
+    |}]
+    |}""".stripMargin.filter(_ >= ' ')
```

Comment:
Interesting way of processing out whitespaces. Gave it some thought, seems correct here (maybe not in a general case where whitespace may occur in some field values), but maybe the correctness is not apparent on the first glance, so a quick comment could help?

Reply:
Please review. Added a warning to all 3 of them.